analyzers/src/SonarAnalyzer.Common/Extensions/CompilationExtensions.cs

```diff
@@ -24,4 +24,17 @@ internal static class CompilationExtensions
 {
     public static INamedTypeSymbol GetTypeByMetadataName(this Compilation compilation, KnownType knownType) =>
         compilation.GetTypeByMetadataName(knownType.FullName);
+
+    public static IMethodSymbol GetTypeMethod(this Compilation compilation, SpecialType type, string methodName) =>
+        (IMethodSymbol)compilation.GetSpecialType(type)
+            .GetMembers(methodName)
+            .SingleOrDefault();
+
+    public static bool IsNetFrameworkTarget(this Compilation compilation) =>
+        // There's no direct way of checking compilation target framework yet (09/2020).
+        // See https://github.com/dotnet/roslyn/issues/3798
+        compilation.ObjectType.ContainingAssembly.Name == "mscorlib";
```

Comment:
This is somewhat overlapping with `NetFrameworkVersionProvider` class
https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Helpers/NetFrameworkVersionProvider.cs

While this extension can live here, the implementation should be in that class to have this kind of ugly detection should live on one place.

Reply:
The implementations can not easily be merged. I created https://github.com/SonarSource/sonar-dotnet/issues/6751